pkg/target/regolib/src.rego

```diff
@@ -315,15 +320,35 @@ matches_namespaces(match) {
 # Always match cluster scoped resources, unless resource is namespace
 matches_namespaces(match) {
   has_field(match, "namespaces")
-  always_match_ns_selectors(match)
+  always_match_ns_selectors
 }
 
 matches_namespaces(match) {
   has_field(match, "namespaces")
-  not always_match_ns_selectors(match)
+  not always_match_ns_selectors
   get_ns_name[ns]
   nss := {n | n = match.namespaces[_]}
-  count({ns} - nss) == 0
+  # count({ns} - nss) == 0
+  # JULIAN - this makes way more sense to me
+  nss[ns]
+}
+
+# Prefix-based matching
+matches_namespaces(match) {
+  has_field(match, "namespaces")
+  not always_match_ns_selectors
+  get_ns_name[ns]
+  wild_nss := wildcard_namespaces(match.namespaces)
+  prefix_glob_match(wild_nss, ns)
+}
+
+wildcard_namespaces(ns_array) = out {
+  out := [ nss | endswith(ns_array[i], "*")
```

Comment:
nit: `[ nss` (remove space or add a space before the closing bracket)

Reply:
done